src/pages/api/stripe/webhook.ts

```diff
@@ -76,12 +91,13 @@ const stripeWebhookHandler = async (
         const previousSubscription = event.data.previous_attributes
         // if the previous attributes have a plan it's an upgrade/downgrade
         if (previousSubscription.plan) {
-          const newPlanAmount =
-            stripeSubscription.items.data[0].plan?.amount || 0
-          const previousPlanAmount = previousSubscription.plan?.amount || 0
+          const previousPlanInterval = previousSubscription.plan?.interval || ''
+          const newPlanInterval =
+            stripeSubscription.items.data[0].plan?.interval || ''
 
-          const state =
-            previousPlanAmount < newPlanAmount ? 'upgrade' : 'downgrade'
+          const state = checkForUpgrade(previousPlanInterval, newPlanInterval)
+            ? 'upgrade'
+            : 'downgrade'
```

Comment:
pulls interval data from stripe and calls `checkForUpgrade`

Reply:
you could return `upgrade` or `downgrade` instead of `true` or `false` from the `checkForUpgrade`?

Reply:
looks like `subscriptionInterval` is being created down below too on line 104

Reply:
> looks like `subscriptionInterval` is being created down below too on line 104

good catch, `newPlanInterval` (on 95) and `subscriptionInterval` (on 104) are basically the same